eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/JpaEventStorageEngine.java

```diff
@@ -105,6 +100,14 @@ protected JpaEventStorageEngine(Builder builder) {
         this.lowestGlobalSequence = builder.lowestGlobalSequence;
         this.gapTimeout = builder.gapTimeout;
         this.gapCleaningThreshold = builder.gapCleaningThreshold;
+
+        this.legacyJpaOperations = new LegacyJpaEventStorageOperations(
+                transactionManager,
+                entityManagerProvider.getEntityManager(),
```

Comment:
The `EntityManagerProvider` can technically return different `EntityManager` instances based on application configuration or moment of execution. It would be better to pass the provider here, not the `EntityManager` at the moment of initialization.
While I don't see any use-cases currently in the framework, it might make a difference.

Reply:
@CodeDrivenMitch you're right, thank you for spotting that :) Fixed.